[PE-D][Tester D] Uncaught AssertionError when adding non-existent student to a group

**Bug Description:**
The command `add_to_group i/E9999999 g/T05` throws an uncaught AssertionError when the provided NUSNETID does not correspond to an existing student.

**Steps to Reproduce:**
1. Ensure no student with ID `E9999999` exists.
2. Execute: `add_to_group i/E9999999 g/T05`
3. Observe the response and terminal.

**Expected Behavior:**
The app should reject the command with a clear message: “Student with NUSNETID E9999999 not found.”

**Actual Behavior:**
The app throws an AssertionError internally:
> `java.lang.AssertionError: Person with given nusnetId should exist in the address book.`  
No user-facing message is shown.

**Screenshots:**
<img width="682" height="88" alt="Image" src="https://raw.githubusercontent.com/nus-cs2103-AY2526S1/ped/master/files/5bcde3b4-622e-406f-938b-85632f798dd5.png" />

**Expected vs Actual:**

| Category | Expected (per UG) | Actual (in App) |
|-----------|-------------------|-----------------|
| Invalid Input Handling | Graceful rejection with message | Uncaught AssertionError thrown |
| UI Feedback | Error displayed to user | No feedback; crash in background |



-------------
Labels: `severity.High` `type.FunctionalityBug`

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -533,7 +533,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 2c. Try to update group id.
 
-    * 2c1. AddressBook shows error: `Group ID cannot be edited directly. Use add_to_group commands instead.`
+    * 2c1. AddressBook shows error and states the correct format.
 
       Use case ends.
 

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -492,7 +492,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 2c. Group ID format is invalid.
 
-    * 2c1. AddressBook shows error message: `Invaid Group ID`.
+    * 2c1. AddressBook shows error message: `Invalid Group ID`.
 
         Use case ends.
 

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -1046,7 +1046,7 @@ testers are expected to do more *exploratory* testing.
    1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
 
    1. Test case: `delete 1`<br>
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message.
 
    1. Test case: `delete 0`<br>
       Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -30,7 +30,7 @@ Given below is a quick overview of main components and how they interact with ea
 
 **Main components of the architecture**
 
-**`Main`** (consisting of classes [`Main`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/Main.java) and [`MainApp`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/MainApp.java)) is in charge of the app launch and shut down.
+**`Main`** (consisting of classes [`Main`](https://github.com/AY2526S1-CS2103T-W11-1/tp/blob/master/src/main/java/seedu/address/Main.java) and [`MainApp`](https://github.com/AY2526S1-CS2103T-W11-1/tp/blob/master/src/main/java/seedu/address/MainApp.java)) is in charge of the app launch and shut down.
 * At app launch, it initializes the other components in the correct sequence, and connects them up with each other.
 * At shut down, it shuts down the other components and invokes cleanup methods where necessary.
 

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -817,7 +817,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 **MSS**
 
-1. User requests to create a new group with a specified GroupName.
+1. User requests to create a new group with a specified GroupId.
 2. Address Book validates the GroupId.
 3. System creates the group.
 4. System shows confirmation message.
@@ -827,17 +827,17 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **Extensions**
 
 * 2a. The GroupId is missing.
-    * 2a1. System shows error message, saying Group Id is missing.
+    * 2a1. System shows error message, saying GroupId is missing.
 
     Use case ends.
 
 * 2b. The GroupId is invalid.
-    * 2c1. System shows error message and indicates the valid format for Group Id.
+    * 2c1. System shows error message and indicates the valid format for GroupId.
 
       Use case ends.
 
 * 2c. The GroupId is a duplicate.
-    * 2b1. System shows error message, saying Group Id already exists
+    * 2b1. System shows error message, saying GroupId already exists
 
       Use case ends.
 

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -466,7 +466,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 **MSS**
 
-1. User requests to add a student by specifying required fields: full name, NUSNET ID, email, Telegram handle ann optional fields: phone number, group ID.
+1. User requests to add a student by specifying required fields: full name, NUSNET ID, email, Telegram handle and optional fields: phone number, group ID.
 
 2. AddressBook validates all fields.
 

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -179,7 +179,7 @@ Format: `edit_student INDEX [n/NAME] [i/NUSNETID]  [t/TELEGRAM] [p/PHONE] [e/EMA
 * Edits the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list. The index **must be a positive integer** 1, 2, 3, …​
 * At least one of the fields is provided to change the person's details.
 * Existing values will be updated to the input values.
-* You can not use this command to change a person's tutorial group. Use the `add_to_group` command instead.
+* You CANNOT use this command to change the tutorial group that this student belongs to. Use the `add_to_group` command instead.
 * For duplicate checking, NUSNET ID, Telegram handle, Phone Number and Email must be unique across all persons in the SoCTAssist.
 * You can use p/ to remove the phone number or e/ to remove the email address of a person by leaving the parameter value empty.
   e.g. `edit_student 2 p/` will remove the phone number of the 2nd person in the displayed person list.
@@ -391,9 +391,9 @@ Adds a student to a tutorial group.
 
 Format: `add_to_group i/NUSNETID g/GROUPID`
 
-* move a student with the specified NUSNET ID to a group with the specified group ID.
+* move a student with the specified NUSNET ID to a new group with the specified group ID.
 * If the specified group does not exist, it will be created.
-* If the group exists, the student will be added to that group.
+* If the group exists, the student will be added to that new group.
 * Student cannot be moved to the same group they are already in; an error message will be shown in such cases.
 * Since a student can only belong to one group at a time, adding them to a new group will remove them from their previous group.
 * The NUSNET ID and group ID **must be valid**. For group ID, refer [here](#creating-a-group--create_group).

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -22,7 +22,7 @@ done faster than traditional Graphical User Interface(GUI) apps while still havi
    **Windows users:** Ensure you have the precise JDK version prescribed [here](https://se-education.org/guides/tutorials/javaInstallationWindows.html).
    **Mac users:** Ensure you have the precise JDK version prescribed [here](https://se-education.org/guides/tutorials/javaInstallationMac.html).
 
-1. Download the latest `.jar` file from [here](https://github.com/AY2526S1-CS2103T-W11-1/tp/releases/tag/v1.4).
+1. Download the latest `.jar` file from [here](https://github.com/AY2526S1-CS2103T-W11-1/tp/releases).
 
 1. Copy the file to the folder you want to use as the _home folder_ for your SoCTAssist.
 

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -27,6 +27,12 @@ done faster than traditional Graphical User Interface(GUI) apps while still havi
 1. Copy the file to the folder you want to use as the _home folder_ for your SoCTAssist.
 
 1. Open a command terminal, using command `cd` + the path to get into the folder you put the jar file in, and use the `java -jar soctassist.jar` command to run the application.<br>
+   Assuming the folder path is `/Users/your_username/Documents/Folder`, the commands will be:<br>
+   Example for Mac/Linux:
+   ```
+   cd /Users/your_username/Documents/Folder
+   java -jar soctassist.jar
+   ```
    A GUI similar to the below should appear in a few seconds. <br>
    ![Ui](images/Ui.png)
 

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -19,6 +19,7 @@ done faster than traditional Graphical User Interface(GUI) apps while still havi
 # Quick Start
 
 1. Ensure you have Java `17` or above installed in your Computer.<br>
+   **Windows users:** Ensure you have the precise JDK version prescribed [here](https://se-education.org/guides/tutorials/javaInstallationWindows.html).
    **Mac users:** Ensure you have the precise JDK version prescribed [here](https://se-education.org/guides/tutorials/javaInstallationMac.html).
 
 1. Download the latest `.jar` file from [here](https://github.com/AY2526S1-CS2103T-W11-1/tp/releases/tag/v1.4).

--- a/docs/diagrams/FindStudentByGroupSequenceDiagram.puml
+++ b/docs/diagrams/FindStudentByGroupSequenceDiagram.puml
@@ -7,9 +7,7 @@ participant ":LogicManager" as LogicManager LOGIC_COLOR
 participant ":AddressBookParser" as AddressBookParser LOGIC_COLOR
 participant ":FindGroupCommandParser" as FindGroupCommandParser LOGIC_COLOR
 participant "d:FindGroupCommand" as FindGroupCommand LOGIC_COLOR
-participant "newGroup:Group" as Group LOGIC_COLOR
 participant "r:CommandResult" as CommandResultSuccess LOGIC_COLOR
-participant "r:CommandResult" as CommandResultFail LOGIC_COLOR
 end box
 
 box Model MODEL_COLOR_T1
@@ -51,7 +49,6 @@ deactivate AddressBookParser
 LogicManager -> FindGroupCommand : execute(m)
 activate FindGroupCommand
 
-alt Group exists in Model
 FindGroupCommand -> Model : updateFilteredPersonList(predicate))
 activate Model
 
@@ -63,19 +60,6 @@ FindGroupCommand -> CommandResultSuccess
 activate CommandResultSuccess
 CommandResultSuccess --> FindGroupCommand
 deactivate CommandResultSuccess
-
-else Group does not exist in Model
-
-create CommandResultFail
-FindGroupCommand -> CommandResultFail
-activate CommandResultFail
-
-CommandResultFail --> FindGroupCommand
-deactivate CommandResultFail
-
-FindGroupCommand --> LogicManager : r
-deactivate FindGroupCommand
-end
 
 [<--LogicManager
 deactivate LogicManager

--- a/src/main/java/seedu/address/model/ModelManager.java
+++ b/src/main/java/seedu/address/model/ModelManager.java
@@ -260,11 +260,14 @@ public class ModelManager implements Model {
     @Override
     public Person getPersonByNusnetIdFullList(Nusnetid nusnetId) throws CommandException {
         requireNonNull(nusnetId);
-        assert hasPerson(nusnetId) : "Person with given nusnetId should exist in the address book.";
-        Person target = this.addressBook.getUniquePersonList()
-                .stream().filter(p -> p.getNusnetid().equals(nusnetId))
-                .findFirst().orElseThrow(() -> new CommandException(MESSAGE_STUDENT_NOT_FOUND));
-        return target;
+        if (hasPerson(nusnetId)) {
+            Person target = this.addressBook.getUniquePersonList()
+                    .stream().filter(p -> p.getNusnetid().equals(nusnetId))
+                    .findFirst().orElseThrow(() -> new CommandException(MESSAGE_STUDENT_NOT_FOUND));
+            return target;
+        } else {
+            throw new CommandException(MESSAGE_STUDENT_NOT_FOUND);
+        }
     }
     @Override
     public boolean hasConsultation(Consultation consultation) {

